File: examples/checkequip.clj
; Check equipment example in Clojure
; Owain Kenway

; Make sure we have the right number of command line arguments.
(if (= (count *command-line-args*) 3)
  (def token (first *command-line-args*))
  ((.println System/err "Run with API key, siteid and roomid as arguments.")(System/exit 1))
)

; Import necessary classes.
(import uclapi.UCLApiConnection)
(import uclapi.Equipment)
(import java.util.Hashtable)

; Put search parameters into a Hashtable.
(def params (Hashtable.))
(.put params "siteid" (second *command-line-args*))
(.put params "roomid" (.get *command-line-args* 2))

; Do our search and print results.
(println (clojure.string/join "\n" (Equipment/searchAPI (UCLApiConnection. token) params)))

File: examples/checkequip.py
# UCL API example in Jython
# This example searches for  for equipment in a room.

# Owain Kenway

import uclapi.UCLApiConnection
import uclapi.Equipment
import java.util.Hashtable

import sys

# Get our key + query.
if len(sys.argv) != 4:
	sys.exit("Run with API key + siteid + roomid as arguments.")

key = sys.argv[1]
params = java.util.Hashtable()
params.put("siteid", sys.argv[2])
params.put("roomid", sys.argv[3])

# Make our connection and get our results.
conn = uclapi.UCLApiConnection(key)
results = uclapi.Equipment.searchAPI(conn, params)

# Print them out.
for a in results:
	print(a.toString())
